web3/constants.py

```diff
@@ -1,7 +1,15 @@
+from eth_typing import (
+    ChecksumAddress,
+    HexAddress,
+    HexStr,
+)
+
 # Constants as Strings
-ADDRESS_ZERO = "0x0000000000000000000000000000000000000000"
-MAX_INT = "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
-HASH_ZERO = "0x0000000000000000000000000000000000000000000000000000000000000000"
+ADDRESS_ZERO = ChecksumAddress(
```

Comment:
@fselmo Might be an opportunity to clean this up. I noticed this constant is defined within the ens module as well, but it does not type it with ChecksumAddress: https://github.com/ethereum/web3.py/blob/d1842bc0a05657b703bb3ce8bbba6fbf0a21d426/ens/constants.py#L16

Does the ens module usage require a separate constant or could these be combined? Other option might be to just keep as is, `ADDRESS_ZERO` for the checksummed address type and `EMPTY_ADDR_HEX` is the non-checksummed type.

Reply:
It would be nice to combine them actually! Good thought.

Reply:
I totally got that from ens constants indeed, I thought it might be like that for a reason

Reply:
Great! I'll help get this merged in. Will push a refactor of the constants along with release notes in a bit.

Thanks for your contribution @Uxio0 !